src/lib.rs

```diff
@@ -0,0 +1,452 @@
+// Copyright (c) 2018 King's College London
+// Created by the Software Development Team <http://soft-dev.org/>
+//
+// The Universal Permissive License (UPL), Version 1.0
+//
+// Subject to the condition set forth below, permission is hereby granted to any
+// person obtaining a copy of this software, associated documentation and/or
+// data (collectively the "Software"), free of charge and under any and all
+// copyright rights in the Software, and any and all patent rights owned or
+// freely licensable by each licensor hereunder covering either (i) the
+// unmodified Software as contributed to or provided by such licensor, or (ii)
+// the Larger Works (as defined below), to deal in both
+//
+// (a) the Software, and
+// (b) any piece of software and/or hardware listed in the lrgrwrks.txt file
+// if one is included with the Software (each a "Larger Work" to which the Software
+// is contributed by such licensors),
+//
+// without restriction, including without limitation the rights to copy, create
+// derivative works of, display, perform, and distribute the Software and make,
+// use, sell, offer for sale, import, export, have made, and have sold the
+// Software and the Larger Work(s), and to sublicense the foregoing rights on
+// either these or other terms.
+//
+// This license is subject to the following condition: The above copyright
+// notice and either this complete permission notice or at a minimum a reference
+// to the UPL must be included in all copies or substantial portions of the
+// Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+extern crate elf;
+extern crate byteorder;
+
+mod errors;
+#[macro_use]
+mod util;
+
+use std::path::Path;
+use std::io::Cursor;
+use byteorder::{NativeEndian, ReadBytesExt};
+use errors::{SMParserError, SMParserResult};
+use util::{cursor_skip, cursor_align8, cursor_from_elf};
+
+// We only support this version of the stackmap header for now.
+const STACKMAP_VERSION: u8 = 3;
+
+// Sizes in bytes.
+const SIZE_STACK_SIZE_ENTRY: u8 = 24;
+const SIZE_CONSTANT_ENTRY: u8 = 8;
+const SIZE_LOC_ENTRY: u8 = 12;
+const SIZE_LIVEOUT_ENTRY: u8 = 4;
+
+/// Offsets into the stackmap section.
+const OFFS_STACK_SIZE_ENTRIES: u64 = 16;
+
+/// Represents a single stackmap record entry.
+#[derive(Debug, Eq, PartialEq)]
+pub struct SMRec {
+    id: u64,            // Stackmap ID.
+    offset: u32,        // Stackmap offset from start of containing func.
+}
+
+impl SMRec {
+    /// Get the stackmap ID.
```

Comment:
Later on we use the idiom "Returns the stackmap ID" (which I personally prefer).

Reply:
Works for me.